tests/unit/verticals/sca/collection/test_filter_ignored.py

```diff
@@ -0,0 +1,129 @@
+from itertools import chain
+from typing import List
+
+import pytest
+from pygitguardian.sca_models import (
+    SCALocationVulnerability,
+    SCAVulnerability,
+    SCAVulnerablePackageVersion,
+)
+
+from ggshield.verticals.sca.collection.filter_ignored import (
+    filter_unignored_location_vulnerabilities,
+    filter_unignored_vulnerabilities,
+)
+
+
+BASE_VULN_INFO = {
+    "severity": "HIGH",
+    "summary": "Vulnerability summary",
+    "cve_ids": ["CVE-0000-0000"],
+    "created_at": None,
+}
+
+
+@pytest.fixture
+def ignored_vulns() -> List[SCAVulnerability]:
+    return [
+        SCAVulnerability.from_dict(
+            {
+                **BASE_VULN_INFO,
+                "status": "IGNORED",
+                **additional_info,
+            }
+        )
+        for additional_info in (
+            {"identifier": "id_0001"},
+            {"identifier": "id_0002", "ignore_reason": "ggshield"},
+            {
+                "identifier": "id_0003",
+                "url": "https://github.com/owner/repo/path",
+                "ignored_until": "2020-01-01T00:00:00",
+                "ignore_reason": "some reason",
+                "ignore_comment": "some comment",
+            },
+        )
+    ]
+
+
+@pytest.fixture
+def non_ignored_vulns() -> List[SCAVulnerability]:
+    return [
+        SCAVulnerability.from_dict(
+            {
+                **BASE_VULN_INFO,
+                **additional_info,
+            }
+        )
+        for additional_info in (
+            {"identifier": "id_1001"},
+            {"identifier": "id_1002", "status": "TRIGGERED"},
+            {"identifier": "id_1003", "url": "https://github.com/owner/repo/path"},
+        )
+    ]
+
+
+def generate_package_vuln(*args: List[SCAVulnerability]) -> SCAVulnerablePackageVersion:
+    return SCAVulnerablePackageVersion(
+        package_full_name="package_name",
+        version="1.2.3",
+        ecosystem="ecosystem",
+        dependency_type="dependency_type",
+        vulns=args,
+    )
+
+
+def test_filter_unignored_vulnerabilities(
+    ignored_vulns: List[SCAVulnerability], non_ignored_vulns: List[SCAVulnerability]
+):
+    """
+    GIVEN   a list of vulnerabilities
+    WHEN    calling filter_unignored_vulnerabilities
+    THEN    only non ignored vulnerabilities are returned
+    """
+    vulns = list(chain(*zip(ignored_vulns, non_ignored_vulns)))
+    expected_vulns = [vuln for vuln in vulns if vuln.status != "IGNORED"]
+    found_vulns = filter_unignored_vulnerabilities(vulns)
+    assert found_vulns == expected_vulns
```

Comment:
Are we sure the order of both lists will always be the same ? Otherwise, we may want to convert them to `set` in the assertion.

Reply:
Due to the implementation, the list order will be conserved. Also it seems I can't cast to set as the objects are non-serializable.